internal/overlord/servstate/manager.go

```diff
@@ -513,6 +513,38 @@ func (m *ServiceManager) CheckFailed(name string) {
 	}
 }
 
+// SetServiceArgs sets the service arguments provided by "pebble run --args"
+// to their respective services. It adds a new layer in the plan, the layer
+// consisting of services with commands having their arguments changed.
+func (m *ServiceManager) SetServiceArgs(serviceArgs map[string][]string) error {
```

Comment:
Let's try to add a test for this in `manager_test.go` to ensure we're exercising and testing this code.

Reply:
Done, thanks!